#Homework 2 for Data analysis with python <br>
#Importing the wickets.csv file in jupyter notebook

In [923]:
import pandas as pd
import numpy as np

In [924]:
import os

In [925]:
#Changing working directory
os.chdir(r'H:\EMK_Data analysis with python')

In [926]:
#Read 'csv' file
df = pd.read_csv('Wickets.csv',encoding = 'unicode_escape')
display(df.head())

Player       Span   Mat  Inns  Balls   Runs  Wkts     BBI  \
0  M Muralitharan (ICC/SL)  1992-2010   133   230  44039  18180   800  Sep-51   
1           SK Warne (AUS)  1992-2007   145   273  40705  17995   708  Aug-71   
2        JM Anderson (ENG)  2003-2021  164*   304  35079  16575   623  Jul-42   
3         A Kumble (INDIA)  1990-2008   132   236  40850  18355   619  Oct-74   
4         GD McGrath (AUS)  1993-2007   124   243  29248  12186   563  24-Aug   

      BBM    Ave  Econ    SR   5  10  
0  16/220  22.72  2.47  55.0  67  22  
1  12/128  25.41  2.65  57.4  37  10  
2  Nov-71  26.60  2.83  56.3  30   3  
3  14/149  29.65  2.69  65.9  35   8  
4  27-Oct  21.64  2.49  51.9  29   3

##Ques 1:Remove the columns BBI and BBM.

In [927]:
#Removing BBI and BBM column
df = df.drop(['BBI', "BBM"], axis=1)
display(df.head())

Player       Span   Mat  Inns  Balls   Runs  Wkts    Ave  \
0  M Muralitharan (ICC/SL)  1992-2010   133   230  44039  18180   800  22.72   
1           SK Warne (AUS)  1992-2007   145   273  40705  17995   708  25.41   
2        JM Anderson (ENG)  2003-2021  164*   304  35079  16575   623  26.60   
3         A Kumble (INDIA)  1990-2008   132   236  40850  18355   619  29.65   
4         GD McGrath (AUS)  1993-2007   124   243  29248  12186   563  21.64   

   Econ    SR   5  10  
0  2.47  55.0  67  22  
1  2.65  57.4  37  10  
2  2.83  56.3  30   3  
3  2.69  65.9  35   8  
4  2.49  51.9  29   3

In [928]:
print('Number of columns:', df.shape[1])

Number of columns: 12


#Ans 1: BBI and BBM column are removed. Number of column remain is 12.

#Ques 2: How many players played for ICC?

In [929]:
#String spliting 
df_player = df['Player'].str.split("(", expand=True)
display(df_player.head())

#Concatenation of two string
df = pd.concat([df, df_player], axis=1)
df = df.drop('Player', axis=1)

# Renaming the new column
df = df.rename(columns={0: 'Player',
                        1: 'Country'})

#Changing the column order
new_col_sequence = ['Player', 'Span', 'Mat', 'Inns', 'Balls', 'Runs', 'Wkts',
                    'Ave', 'Econ', 'SR', '5', '10','Country']
df = df[new_col_sequence]

#Checking for bowler played for ICC
def icc_check(x):
    if "ICC" in x:
        return "Yes"
    else:
        return "No"
df['played_for_ICC'] = df['Country'].apply(icc_check)
display(df.head())

#Value count 
df['played_for_ICC'].value_counts()


0        1
0  M Muralitharan   ICC/SL)
1        SK Warne      AUS)
2     JM Anderson      ENG)
3        A Kumble    INDIA)
4      GD McGrath      AUS)

Player       Span   Mat  Inns  Balls   Runs  Wkts    Ave  Econ  \
0  M Muralitharan   1992-2010   133   230  44039  18180   800  22.72  2.47   
1        SK Warne   1992-2007   145   273  40705  17995   708  25.41  2.65   
2     JM Anderson   2003-2021  164*   304  35079  16575   623  26.60  2.83   
3        A Kumble   1990-2008   132   236  40850  18355   619  29.65  2.69   
4      GD McGrath   1993-2007   124   243  29248  12186   563  21.64  2.49   

     SR   5  10  Country played_for_ICC  
0  55.0  67  22  ICC/SL)            Yes  
1  57.4  37  10     AUS)             No  
2  56.3  30   3     ENG)             No  
3  65.9  35   8   INDIA)             No  
4  51.9  29   3     AUS)             No

No     74
Yes     5
Name: played_for_ICC, dtype: int64

#Ans: In this dataset 5 player played for ICC.


#Ques 3:How many different countries are present in this dataset? 

In [930]:
#1st removing the "ICC/" and '/ICC' from country to count number of country present in dataset
df['Country'] = df['Country'].str.replace("ICC/", "")
df['Country'] = df['Country'].str.replace("/ICC", "")

#Removing ')' from country #new_keys = df['keys'].str.rstrip('/')
df['Country'] = df['Country'].str.rstrip(')')
display(df.head())

#Now need to check the uniqe value in country column
df['Country'].value_counts()


Player       Span   Mat  Inns  Balls   Runs  Wkts    Ave  Econ  \
0  M Muralitharan   1992-2010   133   230  44039  18180   800  22.72  2.47   
1        SK Warne   1992-2007   145   273  40705  17995   708  25.41  2.65   
2     JM Anderson   2003-2021  164*   304  35079  16575   623  26.60  2.83   
3        A Kumble   1990-2008   132   236  40850  18355   619  29.65  2.69   
4      GD McGrath   1993-2007   124   243  29248  12186   563  21.64  2.49   

     SR   5  10 Country played_for_ICC  
0  55.0  67  22      SL            Yes  
1  57.4  37  10     AUS             No  
2  56.3  30   3     ENG             No  
3  65.9  35   8   INDIA             No  
4  51.9  29   3     AUS             No

AUS      18
ENG      15
INDIA    10
WI        9
SA        8
NZ        7
PAK       7
SL        3
ZIM       1
BDESH     1
Name: Country, dtype: int64

#Answer: Here, in country column 10 unique value are present. Number of country playing test cricket is 10 in this dataset.

#Ques 4: Which player(s) had played for the longest period of time?

In [931]:
#Spilting the string value of 'Span' into two column
df['start_year'] = df['Span'].str[0:4]
df['end_year'] = df['Span'].str[5:]

#Drop the "Span" column
df = df.drop("Span", axis=1)
display(df.head())

#Changing the data type from 'str' to 'int'
df['start_year'] = df['start_year'].astype('int') 
df['end_year'] = df['end_year'].astype('int')
print(df.info())

df['years_played'] = df['end_year'] - df['start_year']
df = df.drop(['start_year', "end_year"], axis=1)
display(df.head())

Player   Mat  Inns  Balls   Runs  Wkts    Ave  Econ    SR   5  10  \
0  M Muralitharan    133   230  44039  18180   800  22.72  2.47  55.0  67  22   
1        SK Warne    145   273  40705  17995   708  25.41  2.65  57.4  37  10   
2     JM Anderson   164*   304  35079  16575   623  26.60  2.83  56.3  30   3   
3        A Kumble    132   236  40850  18355   619  29.65  2.69  65.9  35   8   
4      GD McGrath    124   243  29248  12186   563  21.64  2.49  51.9  29   3   

  Country played_for_ICC start_year end_year  
0      SL            Yes       1992     2010  
1     AUS             No       1992     2007  
2     ENG             No       2003     2021  
3   INDIA             No       1990     2008  
4     AUS             No       1993     2007

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Player          79 non-null     object 
 1   Mat             79 non-null     object 
 2   Inns            79 non-null     int64  
 3   Balls           79 non-null     int64  
 4   Runs            79 non-null     int64  
 5   Wkts            79 non-null     int64  
 6   Ave             79 non-null     float64
 7   Econ            79 non-null     float64
 8   SR              79 non-null     float64
 9   5               79 non-null     int64  
 10  10              79 non-null     int64  
 11  Country         79 non-null     object 
 12  played_for_ICC  79 non-null     object 
 13  start_year      79 non-null     int32  
 14  end_year        79 non-null     int32  
dtypes: float64(3), int32(2), int64(6), object(4)
memory usage: 8.8+ KB
None


Player   Mat  Inns  Balls   Runs  Wkts    Ave  Econ    SR   5  10  \
0  M Muralitharan    133   230  44039  18180   800  22.72  2.47  55.0  67  22   
1        SK Warne    145   273  40705  17995   708  25.41  2.65  57.4  37  10   
2     JM Anderson   164*   304  35079  16575   623  26.60  2.83  56.3  30   3   
3        A Kumble    132   236  40850  18355   619  29.65  2.69  65.9  35   8   
4      GD McGrath    124   243  29248  12186   563  21.64  2.49  51.9  29   3   

  Country played_for_ICC  years_played  
0      SL            Yes            18  
1     AUS             No            15  
2     ENG             No            18  
3   INDIA             No            18  
4     AUS             No            14

In [932]:
#storing the column 'year_played' as ascending order
df.sort_values(by='years_played', ascending = False).head()

Player   Mat  Inns  Balls   Runs  Wkts    Ave  Econ    SR   5  \
21      Imran Khan     88   142  19458   8258   362  22.81  2.54  53.7  23   
55       GS Sobers     93   159  21599   7999   235  34.03  2.22  91.9   6   
9     HMRKB Herath     93   170  25993  12157   433  28.07  2.80  60.0  34   
0   M Muralitharan    133   230  44039  18180   800  22.72  2.47  55.0  67   
2      JM Anderson   164*   304  35079  16575   623  26.60  2.83  56.3  30   

    10 Country played_for_ICC  years_played  
21   6     PAK             No            21  
55   0      WI             No            20  
9    9      SL             No            19  
0   22      SL            Yes            18  
2    3     ENG             No            18

#Answer 4: Imran Khan had played for 21 years which is the longest period as a bolower in the dataset.

#Ques 5: Which player(s) had played for the shortest period of time?

In [933]:
#Sort the year_played column in descending order
df.sort_values(by='years_played', ascending = True).head()

Player  Mat  Inns  Balls  Runs  Wkts    Ave  Econ    SR   5  10  \
44      GP Swann    60   109  15349  7642   255  29.96  2.98  60.1  17   3   
71      K Rabada    47    86   8785  4846   213  22.75  3.30  41.2  10   4   
54    Yasir Shah   46*    84  13607  7248   235  30.84  3.19  57.9  16   3   
61   SJ Harmison    63   115  13375  7192   226  31.82  3.22  59.1   8   1   
72  JR Hazlewood    55   103  11887  5438   212  25.65  2.74  56.0   9   0   

   Country played_for_ICC  years_played  
44     ENG             No             5  
71      SA             No             6  
54     PAK             No             7  
61     ENG            Yes             7  
72     AUS             No             7

#Answer 5: GP Swann had played for 5 years which is the shortest period as a bolower in this dataset.

Ques 6: How many Australian Bowlers are present in this dataset? <br> Ques 7:Is there any Bangladeshi player present in this dataset?


In [934]:
#Value count in the 'Country' column
df['Country'].value_counts()

AUS      18
ENG      15
INDIA    10
WI        9
SA        8
NZ        7
PAK       7
SL        3
ZIM       1
BDESH     1
Name: Country, dtype: int64

#Ans 6: There are 18 Australian bowlers in total present in this dataset. <br>
#Ans 7: Yes. Only one bangladesh bowler present in this dataset.

#Ques 8. Which player had the lowest economy rate?

In [935]:
#Sorting the 'Econ' column in descending order
df.sort_values(by='Econ', ascending = True).head()

Player Mat  Inns  Balls  Runs  Wkts    Ave  Econ    SR   5  10  \
32      LR Gibbs   79   148  27115  8989   309  29.09  1.98  87.7  18   2   
47      R Benaud   63   116  19108  6704   248  27.03  2.10  77.0  16   1   
35  DL Underwood   86   151  21862  7674   297  25.83  2.10  73.6  17   6   
39       BS Bedi   67   118  21364  7637   266  28.71  2.14  80.3  14   1   
68   CV Grimmett   37    67  14513  5231   216  24.21  2.16  67.1  21   7   

   Country played_for_ICC  years_played  
32      WI             No            18  
47     AUS             No            12  
35     ENG             No            16  
39   INDIA             No            13  
68     AUS             No            11

Answer 8: LR Gibbs had the lowest economy rate 1.98.

#Ques 9:Which player had the lowest strike rate?

In [936]:
#Sorting the 'SR' column in descending order
df.sort_values(by='SR', ascending = True).head()

Player Mat  Inns  Balls   Runs  Wkts    Ave  Econ    SR   5  10  \
71      K Rabada   47    86   8785   4846   213  22.75  3.30  41.2  10   4   
7       DW Steyn   93   171  18608  10077   439  22.95  3.24  42.3  26   5   
20  Waqar Younis   87   154  16224   8788   373  23.56  3.25  43.4  22   5   
19   MD Marshall   81   151  17584   7876   376  20.94  2.68  46.7  22   4   
25     AA Donald   72   129  15519   7344   330  22.25  2.83  47.0  20   3   

   Country played_for_ICC  years_played  
71      SA             No             6  
7       SA             No            15  
20     PAK             No            14  
19      WI             No            13  
25      SA             No            10

#Answer 9: K Rabada had the lowest strike rate 41.2 in this dataset.

#Ques 10: Which player had the lowest bowling average?

In [937]:
#Sorting the 'Ave' column in descending order
df.sort_values(by='Ave', ascending = True).head()

Player  Mat  Inns  Balls   Runs  Wkts    Ave  Econ    SR   5  10  \
19  MD Marshall    81   151  17584   7876   376  20.94  2.68  46.7  22   4   
41     J Garner    58   111  13169   5433   259  20.97  2.47  50.8   7   0   
15  CEL Ambrose    98   179  22103   8501   405  20.99  2.30  54.5  22   3   
33   FS Trueman    67   127  15178   6625   307  21.57  2.61  49.4  17   3   
4    GD McGrath   124   243  29248  12186   563  21.64  2.49  51.9  29   3   

   Country played_for_ICC  years_played  
19      WI             No            13  
41      WI             No            10  
15      WI             No            12  
33     ENG             No            13  
4      AUS             No            14

#Answer 10: MD Marshall had the lowest bowling average 20.94